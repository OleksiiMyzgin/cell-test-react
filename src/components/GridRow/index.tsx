import Cell from '../Cell';
import classes from './style.module.css';

import { GridRowType } from '../../types';

interface GridRowProps {
  list: GridRowType;
}

function GridRow({ list }: GridRowProps) {
  return (
    <div className={classes.grid}>
      {list.map((cell, index) => (
        <Cell key={index} cell={cell} />
      ))}
    </div>
  );
}

export default GridRow;
