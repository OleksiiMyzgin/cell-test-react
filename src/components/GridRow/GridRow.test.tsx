import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import GridRow from './index';

describe('GridRow test:', () => {
  const matrix = new Array(50).fill(0);

  test('snapshot', () => {
    const component = renderer.create(<GridRow list={matrix} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('display correct numbers of elements', () => {
    const { container } = render(<GridRow list={matrix} />);
    expect(container.querySelectorAll('.cell').length).toBe(50);
  });
});
