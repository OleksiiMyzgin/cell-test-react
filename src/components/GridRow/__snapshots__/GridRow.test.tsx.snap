// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow test: snapshot 1`] = `
<div
  className="grid"
>
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
  <div
    className="cell white"
  />
</div>
`;
